### PR TITLE
FUSETOOLS-2437 - offer UI action for switching Camel versions

### PR DESCRIPTION
**Pull request by lhein**
Signed-off-by: Lars Heinemann <lhein.smx@gmail.com>

### CODE REVIEW

**Comment by lhein** [comment]
fixed reported issues. working on tests now.

**Comment by lhein** [comment]
@apupier can u test once more? I improved the dialog a bit.

File: editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/actions/ChangeCamelVersionJob.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+/**
+ * @author  lheinema
+ */
+public class ChangeCamelVersionJob extends WorkspaceJob {
+
+	private IProject project;
+	private String newVersion;
+	
+	/**
+	 *
```
[the remark this answers is not in the document]
Comment (apupier) [reply editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/actions/ChangeCamelVersionJob.java]: still empty javadoc

**Comment by lhein** [comment]
Camel Catalog exists since 2.15.0
(https://repo.maven.apache.org/maven2/org/apache/camel/camel-catalog/)
For sure there are earlier Camel versions but we lack the editor support, so why should we allow setting them?

**Comment by apupier** [comment]
> Camel Catalog exists since 2.15.0
> (https://repo.maven.apache.org/maven2/org/apache/camel/camel-catalog/)
> For sure there are earlier Camel versions but we lack the editor support, so why should we allow setting them?

Using a Camel version having a Catalog is not really mandatory. Anyway, the earlier corresponding Fuse version that we support is 2.15.1 so not a big deal

**Comment by apupier** [comment]
What do you think about my earlier comments?
- I have a spinning cursor for several seconds, this is not ideal but can be improved later on.
- another improvement that be done later is to reuse the "verify" button logic to check for existing version

Do you want to create enhancement request? or completely abandon it?

**Comment by lhein** [comment]
Why do u want the verify button. A check is done automatically when clicking OK. The busy cursor is set when starting the validation. That's probably why u see it spinning. I just saw no other possibility here and I did not want to show a progress bar just for that. Remember I set the message of the dialog to "Checking xyz..." so its obvious what happens in background. Just my thoughts. I am open to discussions.

**Comment by apupier** [comment]
> Why do u want the verify button. A check is done automatically when clicking OK.

just to be coherent with the "New Fuse Integration project" wizard.

> The busy cursor is set when starting the validation. That's probably why u see it spinning. I just saw no other possibility here and I did not want to show a progress bar just for that. Remember I set the message of the dialog to "Checking xyz..." so its obvious what happens in background. Just my thoughts. I am open to discussions.

for actions that lasts more than 5-10 seconds, a progress bar is recommended. i was wondering several times if it was frozen or not.

**Comment by lhein** [comment]
Ok, lets track those 2 issues here:
https://issues.jboss.org/browse/FUSETOOLS-2536

**Comment by lhein** [comment]
I am ignoring the latest build failure as its not related to the PR.

**Pull request merged**